### PR TITLE
add option to choose full compile/incremental compile

### PR DESCRIPTION
**Pull request by ansyral**
Signed-off-by: xuzho <xuzho@microsoft.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -179,14 +179,23 @@ export function activate(context: ExtensionContext) {
...
-						return window.withProgress({ location: ProgressLocation.Window }, p => {
-							return new Promise((resolve, reject) => {
-								p.report({ message: 'Compiling workspace...' });
-								const start = new Date().getTime();
-								languageClient.sendRequest(CompileWorkspaceRequest.type, true).then((s: CompileWorkspaceStatus) => {
-									const elapsed = new Date().getTime() - start;
-									const humanVisibleDelay = elapsed < 1000? 1000:0;
+					commands.registerCommand(Commands.COMPILE_WORKSPACE, (isFullCompile) => {
+						return window.withProgress({ location: ProgressLocation.Window }, async p => {
+							if (isFullCompile === undefined) {
```
Comment (andxu) [inline src/extension.ts]: if (isFullCompile) is enough.

File: src/extension.ts [hunk 2]
```diff
@@ -179,14 +179,23 @@ export function activate(context: ExtensionContext) {
...
+								const selection = await window.showQuickPick(['Incremental', 'Full'], {'placeHolder' : 'please choose compile type:'});
+								if (selection === 'Incremental') {
+									isFullCompile = false;
+								} else {
+									isFullCompile = true;
```
Comment (andxu) [inline src/extension.ts]: isFullCompile  = selection === 'Incremental'

File: src/extension.ts [hunk 3]
```diff
@@ -179,14 +179,23 @@ export function activate(context: ExtensionContext) {
...
+								}
+							}
+							p.report({ message: 'Compiling workspace...' });
+							const start = new Date().getTime();
+							return languageClient.sendRequest(CompileWorkspaceRequest.type, isFullCompile).then((s: CompileWorkspaceStatus) => {
+								const elapsed = new Date().getTime() - start;
+								const humanVisibleDelay = elapsed < 1000? 1000:0;
+								return new Promise((resolve, reject) =>
+								{
 									setTimeout(function () { // set a timeout so user would still see the message when build time is short
```
Comment (andxu) [inline src/extension.ts]: function should be replaced with lamda

File: src/extension.ts [hunk 4]
```diff
@@ -179,14 +179,23 @@ export function activate(context: ExtensionContext) {
...
+								const selection = await window.showQuickPick(['Incremental', 'Full'], {'placeHolder' : 'please choose compile type:'});
+								if (selection === 'Incremental') {
+									isFullCompile = false;
+								} else {
+									isFullCompile = true;
+								}
+							}
+							p.report({ message: 'Compiling workspace...' });
+							const start = new Date().getTime();
+							return languageClient.sendRequest(CompileWorkspaceRequest.type, isFullCompile).then((s: CompileWorkspaceStatus) => {
```
Comment (andxu) [inline src/extension.ts]: s => status

**Comment by ansyral** [comment]
@fbricon @gorkem Could you help view the PR?

**Comment by ansyral** [comment]
@fbricon nice catch! I'll add one more check

**Pull request merged**